modules/ROOT/pages/indexes/indexes-for-full-text-search.adoc

```diff
@@ -0,0 +1,394 @@
+
+The `score` column represents how well the index thinks that the entry matches the given query.
+Thus, in addition to any exact matches, full-text indexes return _approximate_ matches to a given query.
+This is possible because both the property values that are indexed, and the queries to the index, are processed through the analyzer such that the index can find data entities which do not exactly match the provided `STRING`.
+
+The `score` results are always returned in _descending score order_, where the best matching result entry is put first.
+
+This query uses the `db.index.fulltext.queryRelationships` to query the previously created `communications` full-text index for any `message` containing "meeting":
+
+.Query a full-text index for a relationship property
+[source, cypher]
+----
+CALL db.index.fulltext.queryRelationships("communications", "meeting") YIELD relationship, score
+RETURN type(relationship), relationship.message, score
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+| type(relationship) | relationship.message | score
+
+| "EMAILED" | "I have booked a team meeting tomorrow." | 0.3239005506038666
+
+3+d|Rows: 1
+
+|===
+
+To only obtain exact matches, quote the `STRING` you are searching for:
+
+.Query a full-text index for exact matches
+[source, cypher]
+----
+CALL db.index.fulltext.queryNodes("namesAndTeams", '"Nils-Erik"') YIELD node, score
+RETURN node.name, score
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| node.name | score
+| "Nils-Erik Karlsson" | 0.7588480710983276
+2+d|Rows: 1
+|===
+
+Lucene also allows the use of logical operators, such as `AND` and `OR`, to search for terms:
+
+.Query a full-text index using logical operators
+[source, cypher]
+----
+CALL db.index.fulltext.queryNodes("namesAndTeams", 'nils AND kernel') YIELD node, score
+RETURN node.name, node.team, score
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| node.name | node.team | score
+| "Nils-Erik Karlsson" | "Kernel" | 0.723090410232544
+3+d|Rows: 1
+
+|===
+
+It is possible to limit the search to specific properties, by putting the property name and a colon in front of the text being searched for.
+
+.Query a full-text index for specific properties
+[source, cypher, indent=0]
+----
+CALL db.index.fulltext.queryNodes("namesAndTeams", 'team:"Operations"') YIELD node, score
+RETURN node.name, node.team, score
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| node.name | node.team | score
+| "Nils Johansson" | "Operations" | 0.21363800764083862
+| "Maya Tanaka"   | "Operations" | 0.21363800764083862
+3+d|Rows: 2
+
+|===
+
+A complete description of the Lucene query syntax can be found in the link:https://lucene.apache.org/core/8_2_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description[Lucene documentation].
+
+
+[[text-array-properties]]
+=== Text array properties
```

Comment:
We should refer to "text arrays" as "list of strings", both because `list` is the proper term for the cypher type (and there is a difference in computer science between arrays and lists), and because a "text array" is not a standard term that rings a bell.